core/src/main/java/org/eclipse/hono/config/AbstractConfig.java

```diff
@@ -290,6 +295,42 @@ public final void setKeyPath(final String keyPath) {
         this.keyPath = keyPath;
     }
 
+    /**
+     * Specify the format of the trust store explicitly.
+     * 
+     * @param trustStoreFormat The format to use when reading the trust store, may be {@code null} to trigger auto detection.
+     */
+    public void setTrustStoreFormat(FileFormat trustStoreFormat) {
+        this.trustStoreFormat = trustStoreFormat;
+    }
+
+    /**
+     * Get the specified format of the trust store.
+     * 
+     * @return The format or {@code null} if auto-detection should be tried.
+     */
+    public FileFormat getTrustStoreFormat() {
+        return trustStoreFormat;
+    }
+
+    /**
+     * Specify the format of the key material explicitly.
+     * 
+     * @param keyFormat The format to use when reading the key material, may be {@code null} to trigger auto detection.
+     */
+    public void setKeyFormat(FileFormat keyFormat) {
```

Comment:
would it probably be sufficient to extend the PEM pattern with additional suffixes as e.g. used by Openshift for the key and cert files?

Reply:
Well for OpenShift that will work. But there will probably always be one case which we didn't think of. Allowing the user to manually specify it seems like a way to fix it once and for all.